rust/numaflow-core/Cargo.toml

```diff
@@ -47,13 +49,19 @@ futures = "0.3.30"
 pin-project = "1.1.5"
 rand = "0.8.5"
 async-nats = "0.38.0"
+tracing-subscriber = "0.3.19"
```

Comment:
`numaflow-core` is a library crate, we initialize `tracing-subscriber` only in the binary crate